### PR TITLE
feat: add docs and bolg post about i18next

### PR DESCRIPTION
**Pull request by Anty0**

### CODE REVIEW

File: blog/2024-09-19-i18next-native-format-export-import-support.mdx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  }}
+/>
+
+We are excited to announce that Tolgee now supports the [i18next format](https://www.i18next.com/), a widely-used solution for handling translations in JavaScript-based localization projects.
+
+<!--truncate-->
+
+One of the formats still needing to be added to Tolgee was the i18next format. Not anymore! The support for the i18next format has landed in Tolgee. You can now import and export your localization data in the i18next format as JSON files.
+
+While only a small subset of i18next features are supported, this should be enough to allow basic usage of the format and simplify migration of your existing projects to Tolgee.
```
Comment (JanCizmar) [inline blog/2024-09-19-i18next-native-format-export-import-support.mdx:31]:
```suggestion
When conversion to ICU placeholders is enabled, Tolgee will automatically convert most of the cases. However, some cases are not supported due to missing alternatives in the ICU. In that case, your i18next parameters will be kept untouched so Tolgee can export the same dataset as when imported. 
```

File: platform/formats/i18next_json.mdx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+---
+id: i18next_json
+title: i18next JSON
+sidebar_label: i18next JSON
+---
+
+i18next is a library and format commonly used for handling translations in JavaScript-based projects. Translations are usually stored as part of JavaScript code or in `.json` files. Tolgee supports importing the `.json` representation of translation mapping, but not all i18next features are supported.
```
Comment (JanCizmar) [inline platform/formats/i18next_json.mdx:7]:
```suggestion
i18next is a library and format commonly used for handling translations in JavaScript-based projects. Translations are usually stored as part of JavaScript code or in `.json` files. Tolgee supports importing the `.json` representation of translation mapping.
```

We want the users to start to use it, not to scare them.

File: platform/formats/index.mdx [hunk 3]
```diff
@@ -82,3 +82,8 @@ With Tolgee, you're able to import and export .arb files. [Read more](flutter_ar
 ## Ruby on Rails (.yaml)
 Ruby YAML is structured in YAML format with array support specific to Ruby on Rails applications.
 [Read more](ruby_yaml.mdx).
+
+## i18next (.json)
+
+The i18next library uses custom format to represent localizations. With Tolgee, you're able to import and export json files with this format, but not all features are supported.
```
Comment (JanCizmar) [inline platform/formats/index.mdx:88]: In all of the other formats, we specify where it's used we don't scare the users about not supported features on the index page.

File: platform/formats/index.mdx [hunk 4]
```diff
@@ -82,3 +82,8 @@ With Tolgee, you're able to import and export .arb files. [Read more](flutter_ar
...
+We recommend using ICU plugin for i18next and exporting translations in that format instead as more features are supported for the ICU format. [Read more](i18next_json.mdx)
```
Comment (JanCizmar) [inline platform/formats/index.mdx:89]: We don't need to do the recommendation here.

Comment (JanCizmar) [inline platform/formats/index.mdx:89]: We don't need to do the recommendation here.

File: platform/formats/i18next_json.mdx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  GenericMessageAndPlaceholderConversion
+} from "../../src/component/docs/formats/GenericMessageAndPlaceholderConversion";
+
+<GenericMessageAndPlaceholderConversion formatName="i18next JSON"
+                                        placeholderExamples={<>
+                                          <code>{'{{'}name{'}}'}</code>, <code>{'{{'}count, number{'}}'}</code></>} />
+
+### Placeholder conversion specification
+
+Only specifiers specified in the table below are supported. Tolgee doesn't support additional flags and modifiers and won't convert such placeholders.
```
Comment (JanCizmar) [inline platform/formats/i18next_json.mdx:50]: You are not mentioning how it behaves when unsupported placeholder is imported and exported. We need to asure users that when they import some data, the same data are exported. That's important.

**Review by JanCizmar — CHANGES_REQUESTED** [review]

**Review by JanCizmar — APPROVED** [review]

File: blog/2024-09-19-i18next-native-format-export-import-support.mdx (hunk 1 above)
Comment (Anty0) [reply blog/2024-09-19-i18next-native-format-export-import-support.mdx]: Changed!

File: platform/formats/index.mdx (hunk 3 above)
Comment (Anty0) [reply platform/formats/index.mdx]: It should be fixed.

File: platform/formats/i18next_json.mdx (hunk 5 above)
Comment (Anty0) [reply platform/formats/i18next_json.mdx]: True. I've added something to explain it better. Hopefully, it will be understandable to users.

**Pull request merged**